catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -628,4 +750,77 @@ static Path formalizePath(Path path, Configuration configuration) throws IOExcep
   void setProxyPlugin(HadoopProxyPlugin hadoopProxyPlugin) {
     this.proxyPlugin = hadoopProxyPlugin;
   }
+
+  private UserGroupInformation getUGIByIdent(Map<String, String> properties, NameIdentifier ident) {
+    KerberosConfig kerberosConfig = new KerberosConfig(properties);
+    if (kerberosConfig.isKerberosAuth()) {
+      // We assume that the realm of catalog is the same as the realm of the schema and table.
+      String keytabPath =
+          String.format(
+              GRAVITINO_KEYTAB_FORMAT, catalogInfo.id() + "-" + ident.toString().replace(".", "-"));
+      initKerberos(keytabPath, properties, new Configuration(), ident);
+    }
+    // If the kerberos is not enabled (Simple mode), we will use the current user
+    return getUserBaseOnNameIdentifier(ident);
+  }
+
+  private UserGroupInformation getUserBaseOnNameIdentifier(NameIdentifier nameIdentifier) {
+    UserInfo userInfo = getNearestUserGroupInformation(nameIdentifier);
+    if (userInfo == null) {
+      try {
+        return UserGroupInformation.getCurrentUser();
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to get login user", e);
+      }
+    }
+
+    UserGroupInformation ugi = userInfo.loginUser;
+    boolean userImpersonation = userInfo.enableUserImpersonation;
+    if (userImpersonation) {
+      String proxyKerberosPrincipalName = PrincipalUtils.getCurrentUserName();
+      if (!proxyKerberosPrincipalName.contains("@")) {
+        proxyKerberosPrincipalName =
+            String.format("%s@%s", proxyKerberosPrincipalName, kerberosRealm);
+      }
+
+      ugi = UserGroupInformation.createProxyUser(proxyKerberosPrincipalName, ugi);
+    }
+
+    return ugi;
+  }
+
```

Comment:
@Nullable.

Reply:
Can you clarify the comment? I can't get it.